### ISSUE TITLE
Switch back to the official etherpad-lite branch

### ISSUE THREAD

**Issue by alangecker** [bigbluebutton/bigbluebutton#11069]
## Status Quo
Currently BBB uses a [etherpad-lite fork](https://github.com/bigbluebutton/etherpad-lite) from 2018 which is more or less equivalent to 1.8.0 because upstream changes got merged into it with https://github.com/bigbluebutton/etherpad-lite/pull/3 (2019-12-18), but with some additional small changes for BBB like some theming, extra informations in events and the usage of session tokens.

## The Problems
- Changes are quite difficult to track because the git history diverges quite a lot from the official etherpad branch.
- Maintaining the fork takes quite a lot of effort, because merging upstream patches are not trivial.
- Outdated. multiple security patches since 1.8.0 are not included:
  ```
  * SECURITY: Fix authentication and authorization bypass vulnerabilities
  * IMPORTANT SECURITY: Rate limit Commits when env=production
  * SECURITY: Non completed uploads no longer crash Etherpad
  * SECURITY: Log authentication requests
  * SECURITY: bumped many dependencies. At the time of the release, this version has 0 reported vulnerabilities by npm audit
  * SECURITY: never send referrer when opening a link
  * SECURITY: rate limit imports and exports
  * SECURITY: do not allow pad import if a user never contributed to that pad
  * SECURITY: expose configuration parameter for limiting max import size
  ```

## Proposal
For my  BigBlueButton docker project [alangecker/bigbluebutton-docker](https://github.com/alangecker/bigbluebutton-docker/) I've decided to go a different path which I would like to propose also for the official BBB project:

- **Use the official etherpad branch**
   ..and throw away the [bigbluebutton/etherpad-lite](https://github.com/bigbluebutton/etherpad-lite) fork
- **Merge necessary changes upstream**
   [ether/etherpad-lite#4425: include more data within padUpdate hook](https://github.com/ether/etherpad-lite/pull/4425)
  this is necessary for the closed captions and got included in today's etherpad 1.8.7 release.
- **Use a skin for the consistent look**
   https://github.com/alangecker/bbb-etherpad-skin
- **Use a plugin for further customizations**
   https://github.com/alangecker/bbb-etherpad-plugin

## Steps
instead of using the fork, building the `bbb-etherpad` package could be archived by: 
1. get etherpad-lite >= 1.8.7
2. add plugins
   ```
   npm install  \
    ep_delete_after_delay_lite \
    git+https://git@github.com/pedrobmarin/ep_redis_publisher.git \
    git+https://git@github.com/alangecker/bbb-etherpad-plugin.git
    ```
3. add [skin](https://github.com/alangecker/bbb-etherpad-skin) to `etherpad-lite/src/static/skins/bigbluebutton`
4. make following changes to `settings.json`
    ```json
    {
        "skinName": "bigbluebutton",
        "cookie": {
            "sameSite": "None"
        }
    }
    ```

I would be happy to transfer the ownership of the repositories to the [bigbluebutton team](https://github.com/bigbluebutton)!

**Comment by ffdixon** [comment]
Thanks for this feedback -- we're planning on using a newer version of Etherpad for 2.3-dev.  We'll take a look at your proposal.

**Comment by pedrobmarin** [comment]
We should definitely switch to this approach.

I've been following @alangecker's work on bbb related projects and they are amazing. His docker and packages projects are gems.

**Comment by antobinary** [comment]
Resolved as part of 2.3-alpha6, see release notes - packaging https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.3-alpha-6

**Comment by alangecker** [comment]
coolio :)
@antobinary Should I move the ownership of the packages to the [bigbluebutton organization](https://github.com/bigbluebutton)?
If yes, would be nice if someone with "owner" rights could get in touch with me to handle that transfer :) -> me@chandi.it
